lib/robot.js

```diff
@@ -83,20 +86,24 @@ class Robot {
    *   // An issue was just opened.
    * });
    */
-  on (event, callback) {
-    if (event.constructor === Array) {
-      event.forEach(e => this.on(e, callback))
+  on (webhook, callback) {
+    if (webhook.constructor === Array) {
+      webhook.forEach(e => this.on(e, callback))
       return
     }
 
-    const [name, action] = event.split('.')
+    const [name, action] = webhook.split('.')
 
     return this.events.on(name, async event => {
       if (!action || action === event.payload.action) {
         const log = this.log.child({name: 'event', id: event.id})
+        let github = null
 
         try {
-          const github = await this.auth(event.payload.installation.id, log)
+          if (!(webhook in breakingEvents)) {
+            github = await this.auth(event.payload.installation.id, log)
+          }
```

Comment:
Per the suggestion of just returning an unauthenticated client above, it might make sense for now to just pass on the client from `this.auth()` with no arguments, which is a client that is authenticated as the app instead of as the installation.